src/overrides/venia-ui/components/Button/button.js

```diff
@@ -52,7 +50,8 @@ const Button = props => {
             {...buttonProps}
             {...restProps}
         >
-            <span className={classes.content}><img src="/src/overrides/venia-ui/components/Button/ic_cart.png" alt="shopping card icon"/>{children}</span>
+            <span className={classes.content}><img src="/src/overrides/venia-ui/components/Button/ic_cart.png"
+                                                   alt="shopping card icon"/>{children}</span>
```

Comment:
This Button component is used globally in the whole application, so probably this Icon Cart should be added only for specific buttons in a specific component instead of globally

You might pass the icon as children

Also, for Icons, react-feather package should be used (installed by default), check how it is implemented in other default venia-ui components